packages/react-devtools-timeline/src/CanvasPageContextMenu.js

```diff
@@ -147,16 +150,19 @@ export default function CanvasPageContextMenu({
     if (flamechartStackFrame != null) {
       items.push(
         {
-          onClick: () => copy(flamechartStackFrame.scriptUrl),
+          onClick: withPermissionsCheck({permissions: ['clipboardWrite']}, () =>
+            copy(flamechartStackFrame.scriptUrl),
+          ),
```

Comment:
Do we currently have any `copy` that doesn't check permissions? Feels like every `copy` call should check those i.e. we should have the permission check built-into a custom `copy` function.

Reply:
No, I've double-checked every import of `clipboard-js` that we are using for copying to clipboard.

Ideally, we should stop using this  deprecated npm package, and migrate to `navigator.clipboard.writeText(...)`, but it will be a really tricky case for our current setup.

In short, React DevTools is portaling React roots to other containers, which are mounted in different document, because of the way browser sets up third-party panels for DevTools. This setup leaves the document, which handles user interactions (like click-to-copy), unfocused, and we can't perform `navigator.clipboard.writeText(...)` on unfocused documents.